### ISSUE TITLE
SGD Batch

### ISSUE THREAD

**Issue by flixmk** [ultralytics/yolov5#2540]
## ❔Question
As stated in the code the SGD is used as optimizer. Does changing the batch size in the train.py command change anything regarding that? Cant wrap my mind around how sgd always has a batch size of 1 and being able to select a batch size > 1.
Can someone explain this to me?

**Comment by glenn-jocher** [comment]
@Kraufel not sure I understand what you are asking. Loss is scaled by batch size automatically in YOLOv5 if that is your question.

**Comment by flixmk** [comment]
Yes but when training i can set a batch size. But as far as i know SGD uses a batch size of 1. How do those two things come together?

**Comment by glenn-jocher** [comment]
@Kraufel the optimizer has no relation to batch size, they are completely independent topics. You can train YOLOv5 at any batch size your hardware allows, and you can select any optimizer PyTorch offers for training. The optimizer selection is here, you are free to use any you'd like:
https://github.com/ultralytics/yolov5/blob/e5b0200cd250759c782207160761ca9756300065/train.py#L109-L133
